### ISSUE TITLE
TCP server example

### ISSUE THREAD

**Issue by charleslales** [tmds/Tmds.DBus#37]
Hi,
Is there a way to implement TCP server like it is done here https://bitbucket.org/hindlemail/dbus-sharp/src/2030296a95af6d727cd51d981ba2104f51a140d7/examples/TestServerTcp.cs
I'm trying to establish connection between 2 tools A and B, with B hosting TcpListener and handeling connection from A. The idea is to avoid installing DBus daemon.
Or.. am I wrong?
Thanks in advance for help.
Kind regards,
Charles

**Comment by tmds** [comment]
@charleslales this currently isn't supported. Targeting for next release.

**Comment by charleslales** [comment]
Hi,
Thanks for info, is there any delivery date for it?
I really like the code, clean async await usage.
Furthermore I had to downgrade src to compile to 4.5.2 instead of 4.6 because of user env.
It wasn't so complicated but still may worth a commit / branch.. or?
Charles

**Comment by tmds** [comment]
> Furthermore I had to downgrade src to compile to 4.5.2 instead of 4.6 because of user env.
It wasn't so complicated but still may worth a commit / branch.. or?

Are you running on Windows? Using something older than Win7 SP1?

I guess on the server, the only meaningful operation is making objects available (`RegisterObjectAsync`)? `CreateProxy` wouldn't mean much, since you don't know what peer to target.
Is that right?

**Comment by tmds** [comment]
@charleslales this is implemented in https://github.com/tmds/Tmds.DBus/pull/39. You can find documentation on how to use this feature here: https://github.com/tmds/Tmds.DBus/blob/master/docs/howto.md
You can try this out using the myget feed with Version="0.6.0-*"
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="tmds" value="https://www.myget.org/F/tmds/api/v3/index.json" />
  </packageSources>
</configuration>
```

**Comment by charleslales** [comment]
Cool, thanks! Will test ASAP.
Dev is currently going good, great lib!
Just lately understood that to rise a signal from registered object, you just have to call OnMySignalMeth?.Invoke(args).. nice :)

**Comment by tmds** [comment]
@charleslales is this working for you?
It's nice you are so enthusiastic about the API.
I set as a goal that the interface for exporting objects and creating proxies should be the same. I also wanted to have a single `Connection` class independent of whether it was connecting to a bus or to another peer. I'm glad I could implement this feature request without changing the Connection API (by including it via the constructor options). I like that this feature allows RPC IPC between two applications using D-bus as the serialization protocol.

**Comment by tmds** [comment]
Implemented in #39
